package.json

```diff
@@ -45,7 +45,7 @@
     "typescript": "^4.7.4"
   },
   "dependencies": {
-    "@actions/artifact": "^1.1.0",
+    "@actions/artifact": "^2.1.8",
```

Comment:
As per the npmjs release page (https://www.npmjs.com/package/@actions/artifact), this version might not be supported GHES. 
Test once, if artifact library is supported on GHES

Reply:
Resolved. I had to use two diff versions to work on both GHC and GHE.